[16.0][ADD] stock_location_orderpoint_change_priority

rousseldenis opened the pull request.

rousseldenis committed:
[IMP] stock_location_orderpoint: Add a hook method before procurement run

rousseldenis committed:
[ADD] stock_location_orderpoint_change_priority

mt-software-de commented on stock_location_orderpoint/tests/test_location_orderpoint.py:
```diff
@@ -258,3 +258,69 @@ def test_orderpoint_count(self):
+        replenish_move_manual = self._get_replenishment_move(manual_orderpoint)
+
+        self.assertTrue(replenish_move_manual)
+        self.assertEqual(12.0, replenish_move_manual.product_uom_qty)
```
```suggestion
        self.assertEqual(12.0, replenish_move_manual.product_uom_qty)
        self.assertEqual("0", replenish_move_manual.priority)
```

mt-software-de commented on stock_location_orderpoint_change_priority/models/stock_location_orderpoint.py:
```diff
@@ -0,0 +1,64 @@
+
+    def _find_current_replenishment_moves_for_orderpoint_location(self, products=False):
+        """ """
+        domain = [
+            ("location_dest_id", "child_of", self.location_id.id),
```
This should have a `self.ensure_one()` check. Otherwise `self.location_id.id` could fail. 
```suggestion
        self.ensure_one()
        domain = [
            ("location_dest_id", "child_of", self.location_id.id),
```

mt-software-de commented on stock_location_orderpoint_change_priority/models/stock_location_orderpoint.py:
```diff
@@ -0,0 +1,64 @@
+                    {
+                        "priority": orderpoint.priority,
+                    }
+                )
```
Why is `_find_current_replenishment_moves_for_orderpoint_location` returning the moves grouped by location, when the location is never used? By not grouping it it would ease the code. 
```suggestion
            replenishment_moves = (
                orderpoint._find_current_replenishment_moves_for_orderpoint_location(
                    products=products
                )
            )
            if replenishment_moves:
                replenishment_moves.write(
                    {
                        "priority": orderpoint.priority,
                    }
                )
```

mt-software-de commented on stock_location_orderpoint_change_priority/models/stock_location_orderpoint.py:
```diff
@@ -0,0 +1,64 @@
+
+        We consider the orderpoint has been triggered, so we need
+        """
+        replenishment_moves_to_update = self.env["stock.move"].browse()
```
This shouldn't be defined outside the loop. Otherwise i could write on moves which not belong to the orderpoint.

mt-software-de reviewed (changes requested):
Thx for this change.
Why isn't that implemented in the base addon?
Why hasn't this been implemented in places where we know that the priority might change? For example on `_merge_moves` and on `__prepare_procurements`?

nicolas-delbovier-acsone committed:
[FIX] stock_location_orderpoint: Relocate unit test to correct module

The unit test `test_stock_location_orderpoint_priority_change` was failing
in `stock_location_orderpoint` because the associated logic was moved
to `stock_location_orderpoint_change_priority`.

This commit removes the now-obsolete test from `stock_location_orderpoint`
as an identical test already exists and passes within the
`stock_location_orderpoint_change_priority` module, where the relevant
priority change logic now resides.

nicolas-delbovier-acsone commented:
@mt-software-de Thanks for the review. I will come back later on this. I was jyst fixing this branch as I need it for our client.

nicolas-delbovier-acsone committed:
[IMP] stock_location_orderpoint_change_priority: delay priority update in replenishment logic.

This commit delays the priority update to occur *after* moves have been merged. This ensures that the function merging replenishment moves still has access to the "not yet updated" priorities preventing information loss.

nicolas-delbovier-acsone committed:
[FIX] stock_location_orderpoint: fix date dependend test.

Some test was failing because the code saying "day +1" fails if we are the last day of a month.

nicolas-delbovier-acsone committed:
[REM] stock_location_orderpoint: removes before procurement run hook.

lmignon commented:
@rousseldenis @nicolas-delbovier-acsone Can we close this one in favour of #55.

nicolas-delbovier-acsone commented:
> @rousseldenis @nicolas-delbovier-acsone Can we close this one in favour of #55.

That's fine with me.

jbaudoux commented:
replaced by https://github.com/OCA/stock-logistics-orderpoint/pull/55
